vinteumorg/Floresta
# add mdbook Proposal: Include mdbook for Documentation - Floresta

Labels: enhancement, documentation
  
Description:
I propose integrating mdbook into the Floresta project to enhance our documentation process.

Reasoning:
- Ease of Use: mdbook allows for easy creation and maintenance of documentation directly from Markdown files, which can be managed in version control alongside the codebase.
- Consistency: It provides a uniform look and feel for our documentation, improving user experience.
- Interactive Elements: Features like search functionality, code blocks, and interactive examples can significantly improve the documentation quality.
- Community Standards: Many Rust projects use mdbook, which aligns Floresta with community practices and makes it easier for Rust developers to contribute to or understand our documentation.

Proposal Details:
1. Setup:
    - Add mdbook as a development dependency in Cargo.toml:
```toml
[dev-dependencies]
mdbook = "0.4.14"
```
2. Directory Structure:
    - Create a book directory at the project root:
Using tree-rs
```bash
tree-rs -L 2
```

Looks like this : 
```text
.
├── book
│   ├── 404.html
│   ├── FontAwesome
│   ├── acknowledgments.html
│   ├── ayu-highlight.css
│   ├── benchmarks
│   ├── book.js
│   ├── building
│   ├── clipboard.min.js
│   ├── consensus-implementation.html
│   ├── contrib
│   ├── css
│   ├── elasticlunr.min.js
│   ├── favicon.png
│   ├── favicon.svg
│   ├── fonts
│   ├── fuzz
│   ├── highlight.css
│   ├── highlight.js
│   ├── index.html
│   ├── license.html
│   ├── mark.min.js
│   ├── print.html
│   ├── running
│   ├── searcher.js
│   ├── searchindex.js
│   ├── searchindex.json
│   ├── tests
│   ├── toc.html
│   ├── toc.js
│   ├── tomorrow-night.css
│   └── use_nix
├── book.toml
└── src
    ├── README.md
    ├── SUMMARY.md
    ├── acknowledgments.md
    ├── benchmarks
    ├── building
    ├── consensus-implementation.md
    ├── contrib
    ├── fuzz
    ├── license.md
    ├── running
    ├── tests
    └── use_nix

19 directories, 27 files
```

3. Documentation Content:
    - Write or convert existing documentation into Markdown files under book/src/.
    - The SUMMARY.md file will serve as the table of contents for the book.
4. Build Process:
    - Add a script in Cargo.toml to build the documentation:
5. CI/CD Integration:
    - Update CI scripts to include a step for building documentation with mdBook, possibly deploying it to GitHub Pages or another hosting service. 

Tasks:
- Review existing documentation and convert to Markdown where necessary.
- Set up the initial structure for the mdBook.
- Implement the build script.
- Update CI/CD to automate documentation builds and possibly deployments.

Open Questions:
- Where should the documentation be hosted? GitHub Pages, or another service? @Davidson-Souza mentioned he can help here
- How will we handle versioning of documentation alongside code releases?
- Should this be included in floresta repo (i think so, seems to be common way).

Next Steps:
- Approval from maintainers to proceed with this documentation enhancement.
- Assigning tasks to team members for implementation.
  

Please comment with your thoughts or any additional suggestions. If there's consensus, we can move forward with this enhancement.
